src/bentoml/_internal/server/grpc/servicer/v1/__init__.py

```diff
@@ -81,20 +82,48 @@ async def Call(  # type: ignore (no async types) # pylint: disable=invalid-overr
                         output = await anyio.to_thread.run_sync(api.func, **input_data)
                     else:
                         output = await anyio.to_thread.run_sync(api.func, input_data)
+
                 res = await api.output.to_proto(output)
                 # TODO(aarnphm): support multiple proto fields
                 response = pb.Response(**{api.output._proto_fields[0]: res})
             except BentoMLException as e:
                 log_exception(request, sys.exc_info())
+                if output is not None:
+                    # more descriptive errors if output is available
+                    logger.error(
+                        "Function '%s' has 'input=%s,output=%s' as IO descriptor, and returns 'result=%s'",
```

Comment:
Hm, is this useful information for all of these exceptions? Seems like we only really want to print this when the result type is wrong.

Reply:
I think it is for each exception.